src/main/java/org/opensearch/timeseries/transport/ResultProcessor.java

```diff
@@ -220,14 +224,21 @@ class PageListener implements ActionListener<CompositeRetriever.Page> {
             this.taskId = taskId;
             this.receivedPages = new AtomicInteger();
             this.sentOutPages = new AtomicInteger();
+            this.pagesInFlight = new AtomicInteger();
         }
 
         @Override
         public void onResponse(CompositeRetriever.Page entityFeatures) {
             // start processing next page after sending out features for previous page
             if (pageIterator.hasNext()) {
                 pageIterator.next(this);
+            } else if (config.getImputationOption() != null) {
+                scheduleImputeHCTask();
```

Comment:
Why are we first going inside here and then incrementing the pages inFlight, shouldn't we first increment?

Reply:
yeah, changed

Reply:
I am not 100% sure on this actually, then this `pagesInFlight.get() == 0` will never be reached? I was just thinking of first case also if its 0 it might pass right away

Reply:
When we finished processing all of the inflight requests, pagesInFlight.get() == 0, right?